extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -312,10 +362,12 @@ public void run()
               public ConsumerRecords<byte[], byte[]> call() throws Exception
               {
                 try {
+                  status = Status.READING;
                   return consumer.poll(POLL_TIMEOUT);
                 }
-                finally {
-                  startedReading = true;
+                catch (OffsetOutOfRangeException e) {
+                  status = Status.WAITING;
```

Comment:
The intention was so that if we were hanging around waiting for the offsets to reach our starting offset and we were told to shutdown before we started reading, we could check if the status was waiting and interrupt the thread (since we would be in a retry loop). It may make more sense to just interrupt the thread regardless, even if we are in a reading state. I'll look into it.